examples/plot_autoreject_workflow.py

```diff
@@ -199,6 +197,61 @@
 plt.plot(evoked_bad.times, evoked_bad.data.T * 1e6, 'r', zorder=-1)
 epochs_ar.average().plot(axes=plt.gca())
 
+# %%
+# As a last, optional step, we can do post-doc corrections to be even
+# more stringent in order to favor rejecting epochs over keeping channels
+# marked bad by ``autoreject`` that, based on cross-validation, are
+# considered false positive bad channels.
+
+reject_log = ar.get_reject_log(epochs)
+
+# %%
+# We can observe that, for a few sensors,
+# there are very few green squares and mostly red and blue squares.
+# The autoreject algorithm is keeping the data represented by a
+# red square and considering it a false positive based on the
+# cross-validation scores. For consistency, we can choose to do a
+# post-hoc modification to mark the entire channel for interpolation.
+# We would do this for consistency because whether the square is
+# blue or red depends on the other sensors in that epoch; if there
+# are worse sensors in that epoch, the sensor will be marked red as
+# a false positive and, if not, it will be marked blue. In some cases
+# it may be preferable to interpolate this channel to avoid this effect.
+
+# if more than half bad (red or blue), interpolate
+reject_log.interpolate_bads(interp_thresh=0.5)
+reject_log.plot('horizontal')
+
+# %%
+# Additionally we can chose to drop instead of interpolating any epoch
+# with adjacent bad channels as this will effect the accuracy of the
+# interpolation. Similarly, this should be done on a case-by-case basis
+# based on the montage of sensors and the analysis.
+
+ch_adjacency, ch_names = mne.channels.find_ch_adjacency(epochs.info, 'eeg')
+reject_log.drop_epochs_with_adjacent_channel_interpolation(
+    ch_adjacency, ch_names)
+reject_log.plot('horizontal')
+
+# %%
+# Lastly, if we want to avoid any bad sensors with being left in our data,
+# even on a single epoch, we can drop all epochs with a red square
+# for a bad sensor.
+
+reject_log.drop_epochs_with_bads()
+reject_log.plot('horizontal')
+
+# %%
+# Finally, we can apply the modified reject log to our epoch data. Note
+# that since this example data is not very clean, we have lost most of
+# our epochs; applying these post-hoc corrections should be done as
+# needed in cases where keeping potentially bad data is not desirable.
+
+epochs_ar = ar.transform(epochs, reject_log=reject_log)
+plt.figure()
+plt.plot(evoked_bad.times, evoked_bad.data.T * 1e6, 'r', zorder=-1)
+epochs_ar.average().plot(axes=plt.gca())
```

Comment:
For the tutorial, I would suggest something simple like:

```py
reject_log.labels['EEG 001', :] = 2
```

then pass:

```py
ar.transform(epochs, reject_log=reject_log)
```

I feel this is more broadly applicable and would cover your use cases as well as any future ones. Not super convinced about `drop_epochs_xxx` because they drop so many trials.

Reply:
> The message in the rest of the tutorial seemed to suggest that you need to conserve data, but now you are proposing methods that are more aggressively dropping data, no?

I think this is the big point that justifies the posthoc corrections; they are only more stringent so you'll never decrease the reliability of the results so long as you compensate by collecting more trials.

I think there is also some nuance; we were conserving data with blinks which, in my opinion and definitely in published papers, can be removed effectively by ICA but differences in peak-to-peak amplitude have a much less clear and removable cause. My preferences is to remove independent components that I can confidently say are non-neural but, depending on the data and use, reject data from a likely non-neural source that can't be removed easily.

Reply:
> I feel this is more broadly applicable and would cover your use cases as well as any future ones. Not super convinced about drop_epochs_xxx because they drop so many trials.

There are cases where you might want 20 good trial rather than 50 good and okay trials. Also, an even more rigorous method would be to compare stricter vs less-strict post-hoc corrections/autoreject settings. I think this should be facilitated to validate the method.

Reply:
But autoreject is all about automatic interpolation and saving data ... advocating posthoc methods to drop trials seems a bit like defeating it's purpose. I'm not against adding it for future validation, but maybe the tutorial should focus on API rather than new untested methods.

Reply:
Just logically being more conservative doesn't need validation; you're using a proper subset of the trials you were accepting before so if they were good enough for the first validation, they logically must be good enough in this case. Furthermore validated against what? RANSAC and FASTER? There's no ground truth there either, it's all a house of cards built off whatever algorithm happened to be picked first. I made this PR because for this dataset in the example, there were glaring internal inconsistencies mentioned in the issue. Yes, the validation means something but first and foremost should always be inspecting the dataset and thinking through what makes sense; keeping 50 epochs marked as bad and interpolating 50 epochs out of 110 epochs for one channel just plain doesn't make sense.

Reply:
> Just logically being more conservative doesn't need validation;

if you drop more trials, your average will be more noisy. You need to improve the quality of data, which may or may not improve when bad data is removed. This requires defining metrics of quality and comparing them. It's not as simple as saying "dropping more trials is better".

> Furthermore validated against what? RANSAC and FASTER? There's no ground truth there either, it's all a house of cards built off whatever algorithm happened to be picked first.

This is a question the person validating needs to answer :) The additions must improve upon the state-of-the-art in some measurable way.

Reply:
If you drop more trials you will have fewer epochs unless you collect more data. I think that's the key point, you don't know how much data the user has and `autoreject` is really agnostic to that (i.e. you don't require 40 trials for it to run) so really what `autoreject` does is trial-by-trial validation. If you look at it that way, dropping more trials is better because then there are fewer false positives potentially and you just have to collect more data to make up for the missing trials.

Improving on the state-of-the-art makes sense in a machine learning context like classifying images but it's really not very convincing when it comes to M/EEG trial rejection. There is no benchmark-- you can say a picture has a cat in it and almost everyone will agree but there is no real way to say what a bad trial is other than relative to other channels, which doesn't have a ground truth as far as how different it has to be. I think the main point is that dressing it up as a theoretically grounded approach when it's really just a best-guess approximation is misleading, and the consequence of that--suggesting users take the output of `autoreject` as the truth, rather than tweaking outputs and comparing is counterproductive in my opinion.